{
 "PRed": [
  {
   "PR": {
    "number": 2,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "Compiler",
    "migrator_version": 0
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_version"
   ]
  },
  {
   "PR": {
    "number": 6,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "CompilerRebuild",
    "migrator_version": 1,
    "name": "Python 3.7, GCC 7, R 3.5.1, openBLAS 0.3.2"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 7,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "BlasRebuild",
    "migrator_version": 0,
    "name": "blas2"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 8,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "GFortranOSXRebuild",
    "migrator_version": 0,
    "name": "fortran_compiler_stub-7"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "id": "c3f75a2b-594d-441a-a44a-f5e4a5c5c389",
    "merged_at": "never issued",
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "netcdf-fortran-4.5.x"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 12,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 2,
    "migrator_version": 0,
    "name": "netcdf-fortran-4.5.x"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 18,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "hdf51106"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 19,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "gcc930"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 22,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "libnetcdf480"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 24,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "libnetcdf481"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "number": 29,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "hdf51121"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [],
    "number": 31,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "hdf51122"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [
     {
      "name": "bot-rerun"
     }
    ],
    "number": 32,
    "state": "closed"
   },
   "data": {
    "bot_rerun": 1694370810.1237562,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "netcdf_fortran46"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [
     {
      "name": "bot-rerun"
     }
    ],
    "number": 37,
    "state": "closed"
   },
   "data": {
    "bot_rerun": 1694370810.1239383,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "libnetcdf491"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [
     {
      "name": "bot-rerun"
     }
    ],
    "number": 38,
    "state": "closed"
   },
   "data": {
    "bot_rerun": 1694370810.1240003,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "libnetcdf492"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [
     {
      "name": "bot-rerun"
     }
    ],
    "number": 40,
    "state": "closed"
   },
   "data": {
    "bot_rerun": 1694370810.1240478,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "hdf51141"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [],
    "number": 41,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "hdf51142"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [],
    "number": 42,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "hdf51141"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [],
    "number": null,
    "state": "closed"
   },
   "data": {
    "bot_rerun": false,
    "branch": "v4.1.x",
    "migrator_name": "MigrationYaml",
    "migrator_object_version": 1,
    "migrator_version": 0,
    "name": "hdf51142"
   },
   "keys": [
    "bot_rerun",
    "branch",
    "migrator_name",
    "migrator_object_version",
    "migrator_version",
    "name"
   ]
  },
  {
   "PR": {
    "labels": [
     {
      "name": "bot-rerun"
     }
    ],
    "number": 52,
    "state": "closed"
   },
   "data": {
    "bot_rerun": 1710546673.881334,
    "migrator_name": "OSXArm",
    "migrator_version": 1,
    "name": "arm osx addition"
   },
   "keys": [
    "bot_rerun",
    "migrator_name",
    "migrator_version",
    "name"
   ]
  }
 ],
 "bad": false,
 "pinning_version": "2024.01.25.13.56.16",
 "pre_pr_migrator_attempts": {
  "armosxaddition": 134,
  "gcc930": 1,
  "hdf51140": 272
 },
 "pre_pr_migrator_status": {
  "armosxaddition": "bot error (<a href=\"https://github.com/regro/cf-scripts/actions/runs/9332128920\">bot CI job</a>): main: Traceback (most recent call last):\n  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/auto_tick.py\", line 572, in _run_migrator_on_feedstock_branch\n    migrator_uid, pr_json = run(\n                            ^^^^\n  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/auto_tick.py\", line 216, in run\n    migration_run_data = run_migration(\n                         ^^^^^^^^^^^^^^\n  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/migration_runner.py\", line 63, in run_migration\n    return run_migration_containerized(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/migration_runner.py\", line 157, in run_migration_containerized\n    data = run_container_task(\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/home/runner/work/cf-scripts/cf-scripts/cf-scripts/conda_forge_tick/utils.py\", line 249, in run_container_task\n    raise RuntimeError(\nRuntimeError: Error running migrate-feedstock in container - error DuplicateKeyError('while constructing a mapping', <ruamel.yaml.error.FileMark object at 0x7f8a421fd280>, 'found duplicate key \"conda_build\" with value \"{}\" (original value: \"{}\")', <ruamel.yaml.error.FileMark object at 0x7f8a3c81f2c0>, '\\n                    To suppress this check see:\\n                        http://yaml.readthedocs.io/en/latest/api.html#duplicate-keys\\n                    ', '                    Duplicate keys will become an error in future releases, and are errors\\n                    by default when using the new API.\\n                    ') raised:\ncmd: ['docker',\n 'run',\n '-e',\n 'CF_TICK_IN_CONTAINER=true',\n '--security-opt=no-new-privileges',\n '--read-only',\n '--cap-drop=all',\n '--mount',\n 'type=tmpfs,destination=/tmp,tmpfs-mode=1777,tmpfs-size=6000000000',\n '-m',\n '6000m',\n '--cpus',\n '1',\n '--ulimit',\n 'nofile=1024:1024',\n '--ulimit',\n 'nproc=2048:2048',\n '--rm',\n '-i',\n '--mount',\n 'type=bind,source=/tmp/tmp4smtzlq9,destination=/cf_tick_dir',\n 'ghcr.io/regro/conda-forge-tick:master',\n '/opt/conda/envs/cf-scripts/bin/python',\n '-u',\n '/opt/autotick-bot/docker/run_bot_task.py',\n 'migrate-feedstock',\n '--feedstock-name',\n 'siesta',\n '--default-branch',\n 'main',\n '--existing-feedstock-node-attrs',\n '-',\n '--kwargs',\n '{\\n \"hash_type\": \"sha256\"\\n}',\n '--log-level',\n 'info']\nstdout: {'data': None,\n 'default_branch': 'main',\n 'error': \"DuplicateKeyError('while constructing a mapping', \"\n          \"<ruamel.yaml.error.FileMark object at 0x7f8a421fd280>, 'found \"\n          'duplicate key \"conda_build\" with value \"{}\" (original value: '\n          '\"{}\")\\', <ruamel.yaml.error.FileMark object at 0x7f8a3c81f2c0>, '\n          \"'\\\\n                    To suppress this check \"\n          'see:\\\\n                        '\n          'http://yaml.readthedocs.io/en/latest/api.html#duplicate-keys\\\\n                    '\n          \"', '                    Duplicate keys will become an error in \"\n          'future releases, and are errors\\\\n                    by default '\n          \"when using the new API.\\\\n                    ')\",\n 'feedstock_name': 'siesta',\n 'input_kwargs': '{\\n \"hash_type\": \"sha256\"\\n}',\n 'traceback': 'Traceback (most recent call last):\\n'\n              '  File \"/opt/autotick-bot/docker/run_bot_task.py\", line 120, in '\n              '_run_bot_task\\n'\n              '    data = func(attrs=attrs, **kwargs)\\n'\n              '           ^^^^^^^^^^^^^^^^^^^^^^^^^^^\\n'\n              '  File \"/opt/autotick-bot/docker/run_bot_task.py\", line 308, in '\n              '_migrate_feedstock\\n'\n              '    data = run_migration_local(\\n'\n              '           ^^^^^^^^^^^^^^^^^^^^\\n'\n              '  File '\n              '\"/opt/autotick-bot/conda_forge_tick/migration_runner.py\", line '\n              '241, in run_migration_local\\n'\n              '    data[\"migrate_return_value\"] = migrator.migrate(\\n'\n              '                                   ^^^^^^^^^^^^^^^^^\\n'\n              '  File \"/opt/autotick-bot/conda_forge_tick/migrators/arch.py\", '\n              'line 368, in migrate\\n'\n              '    y = yaml_safe_load(f)\\n'\n              '        ^^^^^^^^^^^^^^^^^\\n'\n              '  File \"/opt/autotick-bot/conda_forge_tick/utils.py\", line 311, '\n              'in yaml_safe_load\\n'\n              '    return ruamel.yaml.YAML(typ=\"safe\", '\n              'pure=True).load(stream)\\n'\n              '           '\n              '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\n'\n              '  File '\n              '\"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/main.py\", '\n              'line 451, in load\\n'\n              '    return constructor.get_single_data()\\n'\n              '           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\n'\n              '  File '\n              '\"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/constructor.py\", '\n              'line 116, in get_single_data\\n'\n              '    return self.construct_document(node)\\n'\n              '           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\n'\n              '  File '\n              '\"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/constructor.py\", '\n              'line 125, in construct_document\\n'\n              '    for _dummy in generator:\\n'\n              '  File '\n              '\"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/constructor.py\", '\n              'line 633, in construct_yaml_map\\n'\n              '    value = self.construct_mapping(node)\\n'\n              '            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\n'\n              '  File '\n              '\"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/constructor.py\", '\n              'line 429, in construct_mapping\\n'\n              '    return BaseConstructor.construct_mapping(self, node, '\n              'deep=deep)\\n'\n              '           '\n              '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\n'\n              '  File '\n              '\"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/constructor.py\", '\n              'line 246, in construct_mapping\\n'\n              '    if self.check_mapping_key(node, key_node, mapping, key, '\n              'value):\\n'\n              '       '\n              '^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\\n'\n              '  File '\n              '\"/opt/conda/envs/cf-scripts/lib/python3.11/site-packages/ruamel/yaml/constructor.py\", '\n              'line 278, in check_mapping_key\\n'\n              '    raise DuplicateKeyError(*args)\\n'\n              'ruamel.yaml.constructor.DuplicateKeyError: while constructing a '\n              'mapping\\n'\n              '  in \"conda-forge.yml\", line 1, column 1\\n'\n              'found duplicate key \"conda_build\" with value \"{}\" (original '\n              'value: \"{}\")\\n'\n              '  in \"conda-forge.yml\", line 13, column 1\\n'\n              '\\n'\n              'To suppress this check see:\\n'\n              '    '\n              'http://yaml.readthedocs.io/en/latest/api.html#duplicate-keys\\n'\n              '\\n'}\n",
  "gcc930": "bot error: Traceback (most recent call last):\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 38, in __call__\n    return cls._cache_[arg]\nKeyError: '>=0*,<0+'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 38, in __call__\n    return cls._cache_[arg]\nKeyError: '<0+'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 38, in __call__\n    return cls._cache_[arg]\nKeyError: '0+'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/home/runner/work/autotick-bot/autotick-bot/cf-scripts/conda_forge_tick/auto_tick.py\", line 943, in main\n    hash_type=attrs.get(\"hash_type\", \"sha256\"),\n  File \"/home/runner/work/autotick-bot/autotick-bot/cf-scripts/conda_forge_tick/auto_tick.py\", line 250, in run\n    solvable, errors, _ = is_recipe_solvable(feedstock_dir)\n  File \"/home/runner/work/autotick-bot/autotick-bot/cf-scripts/conda_forge_tick/mamba_solver.py\", line 433, in is_recipe_solvable\n    additional_channels=additional_channels,\n  File \"/home/runner/work/autotick-bot/autotick-bot/cf-scripts/conda_forge_tick/mamba_solver.py\", line 583, in _is_recipe_solvable_on_platform\n    _solvable, _err = mamba_solver.solve(run_req)\n  File \"/home/runner/work/autotick-bot/autotick-bot/cf-scripts/conda_forge_tick/mamba_solver.py\", line 285, in solve\n    _specs = [_norm_spec(s) for s in specs]\n  File \"/home/runner/work/autotick-bot/autotick-bot/cf-scripts/conda_forge_tick/mamba_solver.py\", line 285, in <listcomp>\n    _specs = [_norm_spec(s) for s in specs]\n  File \"/home/runner/work/autotick-bot/autotick-bot/cf-scripts/conda_forge_tick/mamba_solver.py\", line 82, in _norm_spec\n    m = MatchSpec(myspec)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/match_spec.py\", line 55, in __call__\n    return super(MatchSpecType, cls).__call__(**parsed)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/match_spec.py\", line 177, in __init__\n    self._match_components = self._build_components(**kwargs)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/match_spec.py\", line 403, in _build_components\n    return frozendict(_make_component(key, value) for key, value in iteritems(kwargs))\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/match_spec.py\", line 403, in <genexpr>\n    return frozendict(_make_component(key, value) for key, value in iteritems(kwargs))\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/match_spec.py\", line 417, in _make_component\n    matcher = _implementors[field_name](value)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 40, in __call__\n    val = cls._cache_[arg] = super(SingleStrArgCachingType, cls).__call__(arg)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 497, in __init__\n    vspec_str, matcher, is_exact = self.get_matcher(vspec)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 508, in get_matcher\n    tup = tuple(VersionSpec(s) for s in vspec_tree[1:])\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 508, in <genexpr>\n    tup = tuple(VersionSpec(s) for s in vspec_tree[1:])\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 40, in __call__\n    val = cls._cache_[arg] = super(SingleStrArgCachingType, cls).__call__(arg)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 497, in __init__\n    vspec_str, matcher, is_exact = self.get_matcher(vspec)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 546, in get_matcher\n    self.matcher_vo = VersionOrder(vo_str)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 40, in __call__\n    val = cls._cache_[arg] = super(SingleStrArgCachingType, cls).__call__(arg)\n  File \"/usr/share/miniconda/envs/run_env/lib/python3.7/site-packages/conda/models/version.py\", line 220, in __init__\n    raise InvalidVersionSpec(vstr, \"empty version component\")\nconda.exceptions.InvalidVersionSpec: Invalid version '0+': empty version component\n",
  "hdf51140": "not solvable (<a href=\"https://github.com/regro/cf-scripts/actions/runs/5376158272\">bot CI job</a>): main: ['linux_64_mpimpich: Encountered problems while solving:\\n  - package hdf5-1.10.6-mpi_mpich_h58e33ef_1010 requires libgfortran4 >=7.5.0, but none of the providers can be installed\\n', 'linux_64_mpinompi: Encountered problems while solving:\\n  - package hdf5-1.10.5-nompi_h54c07f9_1110 requires libgfortran4 >=7.5.0, but none of the providers can be installed\\n', 'linux_64_mpiopenmpi: Encountered problems while solving:\\n  - package libnetcdf-4.8.1-mpi_openmpi_h919893c_0 requires hdf5 >=1.10.6,<1.10.7.0a0 mpi_openmpi_*, but none of the providers can be installed\\n', 'osx_64_mpimpich: Encountered problems while solving:\\n  - package netcdf-fortran-4.6.0-mpi_mpich_hea5c964_3 requires libnetcdf >=4.9.1,<4.9.2.0a0 mpi_mpich_*, but none of the providers can be installed\\n', 'osx_64_mpinompi: Encountered problems while solving:\\n  - package netcdf-fortran-4.5.2-nompi_h25f5b61_102 requires hdf5 * nompi_*, but none of the providers can be installed\\n', 'osx_64_mpiopenmpi: Encountered problems while solving:\\n  - package netcdf-fortran-4.5.3-mpi_openmpi_hf49a9dd_5 requires libnetcdf >=4.8.1,<4.8.2.0a0 mpi_openmpi_*, but none of the providers can be installed\\n']"
 },
 "smithy_version": "3.30.4"
}